ES6 Bundle for LZMA_WORKER

umireon opened the pull request:
I have converted lzma_worker.js a ES6 compatible.
In this PR, Rollup.js generates UMD (for Node.js and Browsers) and ES6 bundles.

In addition, this removes use of minify.js and the maintenance will become easier.
The combination of Rollup's dead code removal and UglifyJS's property mangling provides
the same functionality as minify.js without maintaining much code.

umireon committed:
Bundle LZMA_WORKER with Rollup.js

lzma-algo.js is ES6 version of lzma_worker.js.
No algorithms are changed but some properties are quoted for UglifyJS.

lzma-comp.js, lzma-decomp.js, and lzma-codec.js are ES6 equivalents of
lzma-c.js, lzma-d.js, and lzma_worker.js and Rollup produces their minified versions.

umireon committed:
Circumvent false dependencies between functions in Rollup

Some mathematical operation confuses Rollup's dead code removal
algorithm (tree-shaking) and Rollup end up with a bigger bundle than expected.

Problematic increments, decrements and bitwise operations are replaced with
equivalent codes which Rollup can handle well.

Also, $processChunck is split into two variants to avoid including compression
code into decompressor bundle and vice versa.

umireon committed:
Minify with UglifyJS

UglifyJS can compress bundles as small as minify.js does.

umireon committed:
Add browser test cases

iongion commented:
can this be merged ?

ddevault commented:
If this package is abandoned, maybe @umireon would consider forking it and publishing to npm?

mixtur commented:
Merging and publishing this would really be helpful. Especially if it won't break backward compatibility.

umireon committed:
Update rollup.js

umireon committed:
Use terser

umireon committed:
Write generated files in src/

umireon committed:
Update minify.js to use rollup

umireon committed:
Minified:

lzma.js
Original size: 111599 bytes (109.0 KB)
Minified size: 24764 bytes (24.2 KB)
Compression:   4.5065 x smaller
Gzipped size:  9433 bytes (9.2 KB)
Gzipped ratio: 11.8307 x smaller

lzma_worker.js
Original size: 111155 bytes (108.5 KB)
Minified size: 24409 bytes (23.8 KB)
Compression:   4.5539 x smaller
Gzipped size:  9326 bytes (9.1 KB)
Gzipped ratio: 11.9188 x smaller

lzma-c.js
Original size: 85417 bytes (83.4 KB)
Minified size: 18828 bytes (18.4 KB)
Compression:   4.5367 x smaller
Gzipped size:  7464 bytes (7.3 KB)
Gzipped ratio: 11.4439 x smaller

lzma-d.js
Original size: 32223 bytes (31.5 KB)
Minified size: 7746 bytes (7.6 KB)
Compression:   4.1600 x smaller
Gzipped size:  3285 bytes (3.2 KB)
Gzipped ratio: 9.8091 x smaller

umireon committed:
Use iife for uncompressed files

umireon committed:
Minified:

lzma.js
Original size: 111307 bytes (108.7 KB)
Minified size: 24764 bytes (24.2 KB)
Compression:   4.4947 x smaller
Gzipped size:  9433 bytes (9.2 KB)
Gzipped ratio: 11.7997 x smaller

lzma_worker.js
Original size: 110863 bytes (108.3 KB)
Minified size: 24409 bytes (23.8 KB)
Compression:   4.5419 x smaller
Gzipped size:  9326 bytes (9.1 KB)
Gzipped ratio: 11.8875 x smaller

lzma-c.js
Original size: 85125 bytes (83.1 KB)
Minified size: 18828 bytes (18.4 KB)
Compression:   4.5212 x smaller
Gzipped size:  7464 bytes (7.3 KB)
Gzipped ratio: 11.4047 x smaller

lzma-d.js
Original size: 32223 bytes (31.5 KB)
Minified size: 7746 bytes (7.6 KB)
Compression:   4.1600 x smaller
Gzipped size:  3285 bytes (3.2 KB)
Gzipped ratio: 9.8091 x smaller

umireon committed:
Update tests

umireon committed:
Minified:

lzma.js
Original size: 111782 bytes (109.2 KB)
Minified size: 24899 bytes (24.3 KB)
Compression:   4.4894 x smaller
Gzipped size:  9444 bytes (9.2 KB)
Gzipped ratio: 11.8363 x smaller

lzma_worker.js
Original size: 111338 bytes (108.7 KB)
Minified size: 24582 bytes (24.0 KB)
Compression:   4.5292 x smaller
Gzipped size:  9364 bytes (9.1 KB)
Gzipped ratio: 11.8900 x smaller

lzma-c.js
Original size: 111527 bytes (108.9 KB)
Minified size: 24754 bytes (24.2 KB)
Compression:   4.5054 x smaller
Gzipped size:  9428 bytes (9.2 KB)
Gzipped ratio: 11.8293 x smaller

lzma-d.js
Original size: 111524 bytes (108.9 KB)
Minified size: 24756 bytes (24.2 KB)
Compression:   4.5049 x smaller
Gzipped size:  9423 bytes (9.2 KB)
Gzipped ratio: 11.8353 x smaller

umireon commented:
@nmrugg Ensured backward compatibility!

umireon committed:
Update index.js

umireon committed:
xvfb

umireon commented:
@nmrugg Travis CI works on my forked repository,
but it does not work on this repository.
https://travis-ci.org/umireon/LZMA-JS/builds/650269007?utm_medium=notification&utm_source=github_status

umireon committed:
Add index.mjs

umireon committed:
Load lzma.js in index.mjs

umireon committed:
Split LZMA_WORKER

umireon committed:
Minified:

lzma.js
Original size: 111790 bytes (109.2 KB)
Minified size: 24861 bytes (24.3 KB)
Compression:   4.4966 x smaller
Gzipped size:  9428 bytes (9.2 KB)
Gzipped ratio: 11.8572 x smaller

lzma_worker.js
Original size: 111346 bytes (108.7 KB)
Minified size: 24544 bytes (24.0 KB)
Compression:   4.5366 x smaller
Gzipped size:  9350 bytes (9.1 KB)
Gzipped ratio: 11.9087 x smaller

lzma-c.js
Original size: 85425 bytes (83.4 KB)
Minified size: 18804 bytes (18.4 KB)
Compression:   4.5429 x smaller
Gzipped size:  7452 bytes (7.3 KB)
Gzipped ratio: 11.4634 x smaller

lzma-d.js
Original size: 32223 bytes (31.5 KB)
Minified size: 7756 bytes (7.6 KB)
Compression:   4.1546 x smaller
Gzipped size:  3290 bytes (3.2 KB)
Gzipped ratio: 9.7942 x smaller

umireon committed:
Move src/es/*.js to src/*.mjs

umireon committed:
Revert src/lzma.js

umireon committed:
Minified:

lzma.js
Original size: 6604 bytes (6.4 KB)
Minified size: 1696 bytes (1.7 KB)
Compression:   3.8939 x smaller
Gzipped size:  729 bytes (0.71 KB)
Gzipped ratio: 9.0590 x smaller

lzma_worker.js
Original size: 111618 bytes (109.0 KB)
Minified size: 24731 bytes (24.2 KB)
Compression:   4.5133 x smaller
Gzipped size:  9418 bytes (9.2 KB)
Gzipped ratio: 11.8516 x smaller

lzma-c.js
Original size: 85433 bytes (83.4 KB)
Minified size: 18799 bytes (18.4 KB)
Compression:   4.5446 x smaller
Gzipped size:  7451 bytes (7.3 KB)
Gzipped ratio: 11.4660 x smaller

lzma-d.js
Original size: 32231 bytes (31.5 KB)
Minified size: 7751 bytes (7.6 KB)
Compression:   4.1583 x smaller
Gzipped size:  3290 bytes (3.2 KB)
Gzipped ratio: 9.7967 x smaller

umireon committed:
Remove *-min.mjs

umireon committed:
Generate sourcemap

umireon committed:
Minified:

lzma.js
Original size: 6604 bytes (6.4 KB)
Minified size: 1733 bytes (1.7 KB)
Compression:   3.8107 x smaller
Gzipped size:  756 bytes (0.74 KB)
Gzipped ratio: 8.7354 x smaller

lzma_worker.js
Original size: 111658 bytes (109.0 KB)
Minified size: 24775 bytes (24.2 KB)
Compression:   4.5069 x smaller
Gzipped size:  9459 bytes (9.2 KB)
Gzipped ratio: 11.8044 x smaller

lzma-c.js
Original size: 85468 bytes (83.5 KB)
Minified size: 18838 bytes (18.4 KB)
Compression:   4.5370 x smaller
Gzipped size:  7488 bytes (7.3 KB)
Gzipped ratio: 11.4140 x smaller

lzma-d.js
Original size: 32266 bytes (31.5 KB)
Minified size: 7790 bytes (7.6 KB)
Compression:   4.1420 x smaller
Gzipped size:  3326 bytes (3.2 KB)
Gzipped ratio: 9.7011 x smaller

umireon committed:
Update package.json

umireon committed:
Update index.mjs

umireon committed:
Generate src/*-min.mjs for webpack

umireon commented:
@nmrugg https://travis-ci.org/umireon/LZMA-JS/builds/652740261

umireon commented:
@nmrugg src/lzma.js is reverted.

umireon committed:
Export LZMA_WORKER

umireon committed:
Minified:

lzma.js
Original size: 6604 bytes (6.4 KB)
Minified size: 1733 bytes (1.7 KB)
Compression:   3.8107 x smaller
Gzipped size:  756 bytes (0.74 KB)
Gzipped ratio: 8.7354 x smaller

lzma_worker.js
Original size: 111726 bytes (109.1 KB)
Minified size: 24822 bytes (24.2 KB)
Compression:   4.5011 x smaller
Gzipped size:  9469 bytes (9.2 KB)
Gzipped ratio: 11.7991 x smaller

lzma-c.js
Original size: 85536 bytes (83.5 KB)
Minified size: 18885 bytes (18.4 KB)
Compression:   4.5293 x smaller
Gzipped size:  7497 bytes (7.3 KB)
Gzipped ratio: 11.4094 x smaller

lzma-d.js
Original size: 32334 bytes (31.6 KB)
Minified size: 7837 bytes (7.7 KB)
Compression:   4.1258 x smaller
Gzipped size:  3336 bytes (3.3 KB)
Gzipped ratio: 9.6924 x smaller

umireon committed:
remove dist/ from .gitignore

umireon committed:
Fix var to const

umireon commented:
@nmrugg https://travis-ci.org/umireon/LZMA-JS/builds/653253369

umireon committed:
Remove constants from lzma-d.js

umireon committed:
Minified:

lzma.js
Original size: 6604 bytes (6.4 KB)
Minified size: 1733 bytes (1.7 KB)
Compression:   3.8107 x smaller
Gzipped size:  756 bytes (0.74 KB)
Gzipped ratio: 8.7354 x smaller

lzma_worker.js
Original size: 111726 bytes (109.1 KB)
Minified size: 24822 bytes (24.2 KB)
Compression:   4.5011 x smaller
Gzipped size:  9469 bytes (9.2 KB)
Gzipped ratio: 11.7991 x smaller

lzma-c.js
Original size: 85536 bytes (83.5 KB)
Minified size: 18885 bytes (18.4 KB)
Compression:   4.5293 x smaller
Gzipped size:  7497 bytes (7.3 KB)
Gzipped ratio: 11.4094 x smaller

lzma-d.js
Original size: 31061 bytes (30.3 KB)
Minified size: 7485 bytes (7.3 KB)
Compression:   4.1498 x smaller
Gzipped size:  3179 bytes (3.1 KB)
Gzipped ratio: 9.7707 x smaller

umireon commented:
@nmrugg https://travis-ci.org/umireon/LZMA-JS/builds/653947267

umireon committed:
Remove `npm run build` from the browser test

umireon committed:
Remove `npm run build` from test:karma

umireon committed:
Update karma

umireon committed:
Update acorn

bugproof commented:
why this was never released ? anyone could create a fork and push a new npm package since this one is abandoned.
